crates/orchestrator/src/queue/job_queue.rs

```diff
@@ -57,11 +58,42 @@ pub enum WorkerTriggerType {
     UpdateState,
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Clone)]
 pub struct WorkerTriggerMessage {
-    pub(crate) worker: WorkerTriggerType,
+    pub worker: WorkerTriggerType,
 }
 
+impl FromStr for WorkerTriggerType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Proving" => Ok(WorkerTriggerType::Proving),
+            "Snos" => Ok(WorkerTriggerType::Snos),
+            "ProofRegistration" => Ok(WorkerTriggerType::ProofRegistration),
+            "DataSubmission" => Ok(WorkerTriggerType::DataSubmission),
+            "UpdateState" => Ok(WorkerTriggerType::UpdateState),
+            _ => Err(format!("Unknown WorkerTriggerType: {}", s)),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for WorkerTriggerMessage {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
```

Comment:
we should be able to use `serde_json`

Reply:
some error was coming when I was using serde_json that's why I implemented this custom deserializer.

Reply:
do you remember what the error was? maybe we can add it as a comment

Reply:
unable to serialise something error was there.